File: python/p003.py
from euler import isprime


def p003(N):
    """Finds all prime factors of N."""
    largest = 0

    for i in range(2, int(N**0.5+1)):
        # Divide N by its prime factors
        while N % i == 0:
            N /= i
            largest = i

        # Stop loop if N is factored-out
        if N == 1:
            break

    return largest


if __name__ == "__main__":
    print(p003(600851475143))

File: julia/p003.jl
using Primes

function p003(N)
    p = 2
    remainder = N
    largest = 1
    while remainder != 1
        while remainder % p == 0
            remainder ÷= p
            largest = p
        end
        p = Primes.nextprime(p, 2)

        # Shortcut to termination
        if Primes.isprime(remainder)
            largest = remainder
            break
        end
    end
    return largest
end

println(p003(600851475143))
